src/responder/pam/pamsrv_cmd.c

```diff
@@ -2032,11 +2017,19 @@ static bool pam_can_user_cache_auth(struct sss_domain_info *domain,
     errno_t ret;
     bool result = false;
 
-    if (!cached_auth_failed /* don't try cached auth again */
-            && domain->cache_credentials
-            && domain->cached_auth_timeout > 0
-            && pam_is_authtok_cachable(authtok)
-            && pam_is_cmd_cachable(pam_cmd)) {
+    if (cached_auth_failed) {
+        /* Do not retry indefinitely */
+        return false;
+    }
+
+    if (domain->cache_credentials == false
```

Comment:
Nitpick, since you used `if (cached_auth_failed)` above I wonder if you can use `if (!domain->cache_credentials ...` here instead.